test/middleware/CompanyAuthMiddleware.test.ts

```diff
@@ -0,0 +1,155 @@
+        accountClientId: 'MOCK',
+        chsUrl: 'MOCK',
+    };
+
+    const sessionStoreForAuthConfig = {
+        sessionCookieName: '__SID',
+        sessionCookieDomain: 'rebel1.aws.chdev.org',
+        sessionCookieSecureFlag: 'true',
+        sessionTimeToLiveInSeconds: 3600
+    };
+
+    const featureFlag: string = '1';
+
+    it('should call next with an error when session is undefined', async () => {
+
+        const encryptionService = createEncryptionService('resolves');
+
+        const sessionStore = Substitute.for<SessionStore>();
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = { session: undefined } as Request;
+
+        try{
+            await companyAuthMiddleware.handler(request, response, nextFunction);
+        } catch (err) {
+            expect(err.message).to.equal(SESSION_NOT_FOUND_ERROR.message);
+        }
+    });
+
+
+    it('should call next if the user is authorized for company number', async () => {
+
+        const appData = { appeal };
+
+        const encryptionService = createEncryptionService('resolves');
+
+        const sessionStore = Substitute.for<SessionStore>();
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = getRequestSubstitute(appData, appeal.penaltyIdentifier.companyNumber);
+
+        await companyAuthMiddleware.handler(request, response, nextFunction);
+        nextFunction.received(1);
+        response.didNotReceive().redirect(Arg.any());
+
+    });
+
+    it('should redirect if the user is not authorized for company number', async () => {
+
+        const appData = { appeal };
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = getRequestSubstitute(appData, '');
+
+        const encryptionService = createEncryptionService('resolves', 'a_sequence');
+
+        const sessionStore = Substitute.for<SessionStore>();
+
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        await companyAuthMiddleware.handler(request, response, nextFunction);
+        nextFunction.didNotReceive();
+        response.received(1).redirect(Arg.any());
```

Comment:
Is making any redirect a purpose of the company middleware? Saying middleware works fine when any redirect is made is over simplification. Don't you think?